sysdeps/nacl/irt_syscalls.c

```diff
@@ -262,24 +262,32 @@ static int nacl_irt_mutex_create (int *mutex_handle) {
   return 0;
 }
 
-/*
- * Today a mutex handle is just an fd and we destroy it with close.
- * But this might not always be so.
- */
 static int nacl_irt_mutex_destroy (int mutex_handle) {
-  return -NACL_SYSCALL (close) (mutex_handle);
+  int rv = NACL_SYSCALL (mutex_destroy) (mutex_handle);
+  if(rv < 0)
```

Comment:
Pedantic, but we stylistically usually put a space between the if and condition here.